apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts

```diff
@@ -119,11 +155,164 @@ export class GfCreateOrUpdateAccessDialogComponent
     this.unsubscribeSubject.complete();
   }
 
+  private buildFilterObject():
+    | {
+        accountIds?: string[];
+        assetClasses?: string[];
+        holdings?: AssetProfileIdentifier[];
+        tagIds?: string[];
+      }
+    | undefined {
+    const filterValue = this.accessForm.get('filters')
+      ?.value as PortfolioFilterFormValue | null;
+
+    if (
+      !filterValue ||
+      (!filterValue.account &&
+        !filterValue.assetClass &&
+        !filterValue.holding &&
+        !filterValue.tag)
+    ) {
+      return undefined;
+    }
+
+    const filter: {
+      accountIds?: string[];
+      assetClasses?: string[];
+      holdings?: AssetProfileIdentifier[];
+      tagIds?: string[];
+    } = {};
+
+    if (filterValue.account) {
+      filter.accountIds = [filterValue.account];
+    }
+
+    if (filterValue.assetClass) {
+      filter.assetClasses = [filterValue.assetClass];
+    }
+
+    if (filterValue.holding) {
+      filter.holdings = [
+        {
+          dataSource: filterValue.holding.dataSource,
+          symbol: filterValue.holding.symbol
+        }
+      ];
+    }
+
+    if (filterValue.tag) {
+      filter.tagIds = [filterValue.tag];
+    }
+
+    return filter;
+  }
+
+  private loadFilterData() {
+    const existingFilter = this.data.access.settings?.filter;
+
+    this.dataService
```

Comment:
The `user` object (via `userService`) contains the `accounts`, no need to fetch here